app/scripts/controllers/network/network.js

```diff
@@ -124,11 +124,11 @@ export default class NetworkController extends EventEmitter {
     this._infuraProjectId = projectId;
   }
 
-  initializeProvider(providerParams) {
+  async initializeProvider(providerParams) {
```

Comment:
I notice that unit tests are missing for this exact method. `initializeProvider` is called for other tests, but without taking async into consideration. Same goes for `lookupNetwork`.

This should be captured by #972, but it sounds like we wanted to add unit tests _after_ changing this controller. Should that happen before instead so that we can make sure we haven't broken anything?

Reply:
Hmm. Yeah, fair enough. Tests first would help validate the work. I'll investigate how difficult testing this controller more effectively would be.

Reply:
In this case I don't think it's worth the effort to add unit tests prior to making the change, because the change itself is primarily intended to make unit testing easier. Especially given that this code is well covered by e2e tests, writing tests _after_ this PR seems pragmatic to me.

I have added a new issue for tracking unit tests in the extension: https://github.com/MetaMask/metamask-extension/issues/16962
We can make a point to consider this for subsequent network controller changes as well. Unless a change is specifically meant to improve testability, we probably should ensure all future changes to the network controller are covered by unit tests.

Reply:
Okay sounds good!